lib/portlayer/storage/vsphere/image.go

```diff
@@ -641,3 +663,55 @@ func imagesInUse(op trace.Operation, ID string) error {
 
 	return nil
 }
+
+// populate the scratch with minimum OS structure defined in FileForMinOS and DirForMinOS
+func populateMinOS(op trace.Operation, vmdisk *disk.VirtualDisk) error {
```

Comment:
if you name the error return then you can return an error from the deferred function blocks (see https://play.golang.org/p/KySTHY2gM4):
```go
func populateMinOS(op trace.Operation, vmdisk *disk.VirtualDisk) (err error)
```
generally naming returns tends to be confusing, but it's the only way I'm aware of to allow deferred functions to contribute to the returned values.